src/components/QueryEditor.tsx

```diff
@@ -140,6 +140,7 @@ const useSelectedDatabase = (
   query: KustoQuery,
   datasource: AdxDataSource
 ): string => {
+  const defaultDB = useAsync(() => datasource.getDefaultOrFirstDatabase(), [datasource]);
```

Comment:
nit: if you are using this only within the  `useMemo` you can define it there

Reply:
If I do a `useAsync` inside a `useMemo` that results in a warning about using hooks inside `useMemo`.